scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua

```diff
@@ -0,0 +1,46 @@
+-----------------------------------
+-- Area: Bastok Markets [S]
+--  NPC: Kevan
+-- !pos -308.590 -012.000 -094.227 189
+-- Sealed Container NPC
+-----------------------------------
+local ID = require("scripts/zones/Bastok_Markets_[S]/IDs")
+-----------------------------------
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+    local itemID = trade:getItemId()
+    local itemKeyItemMapping =
+    {
+        [2686] = xi.ki.THE_WORDS_OF_DONHU_I,
+        [2687] = xi.ki.THE_WORDS_OF_DONHU_II,
+        [2688] = xi.ki.THE_WORDS_OF_DONHU_III,
+        [2689] = xi.ki.THE_WORDS_OF_DONHU_IV,
+        [2690] = xi.ki.THE_WORDS_OF_DONHU_V,
+        [2691] = xi.ki.THE_WORDS_OF_DONHU_VI,
+        [2692] = xi.ki.THE_WORDS_OF_DONHU_VII,
+        [2693] = xi.ki.THE_WORDS_OF_DONHU_VIII
+    }
+    -- Make sure we're only trading 1 Strongbox at a time.
+    if
+        npcUtil.tradeHasExactly(trade, itemID) and
+        itemKeyItemMapping[itemID]
+    then
+        player:tradeComplete()
+        local keyItem = itemKeyItemMapping[itemID]
+        npc:showText(npc, 13565)
```

Comment:
All of these npc:showText's need to have their text extracted out into the relevant zone's ID file (which you're already requiring at the top).

You have the relevant ID already, so you just need to look up the correct comment from UpdateExtractor (https://github.com/LandSandBoat/UpdateExtractor) so that ID shifts can happen auto-magically

Reply:
Done.